icu4j/main/classes/core/src/com/ibm/icu/impl/locale/LSR.java

```diff
@@ -89,4 +90,109 @@ public boolean equals(Object obj) {
     public int hashCode() {
         return Objects.hash(language, script, region, flags);
     }
+
+    // This method is added only to support encodeToIntForResource()
+    // It only support [a-z]{2,3} and will not work for other cases.
+    private int encodeLanguageToInt() {
+        assert language.length() >= 2;
+        assert language.length() <= 3;
+        assert language.charAt(0) >= 'a';
+        assert language.charAt(0) <= 'z';
+        assert language.charAt(1) >= 'a';
+        assert language.charAt(1) <= 'z';
+        assert language.length() == 2 || language.charAt(2) >= 'a';
+        assert language.length() == 2 || language.charAt(2) <= 'z';
+        return language.charAt(0) - 'a' + 1 +
+               27 * (language.charAt(1) - 'a' + 1) +
+               ((language.length() == 2) ? 0 : 27 * 27 * (language.charAt(2) - 'a' + 1));
+    }
+    // This method is added only to support encodeToIntForResource()
+    // It only support [A-Z][a-z]{3} which defined in UScript and does not work for other cases.
+    private int encodeScriptToInt() {
+        int ret = UScript.getCodeFromName(script);
+        assert ret != UScript.INVALID_CODE;
+        return ret;
+    }
+    // This method is added only to support encodeToIntForResource()
+    // It only support [A-Z]{2}|001|143|419 and does not work for other cases.
+    private int encodeRegionToInt() {
+        assert region.length() >= 2;
+        assert region.length() <= 3;
+        // Do not have enough bits to store the all 1000 possible combination of \d{3}
+        // Only support what is needed in resource- "001", "143" and "419".
+        if (region.length() == 3) {
+            assert region.equals("001") || region.equals("143") || region.equals("419");
```

Comment:
See my comments about this approach in LocaleDistanceMapper.java. It is too fragile and inflexible. I think LocaleDistanceMapper.java should write an array of supported numeric region codes that this method can use for encoding.

Reply:
well do. Changing the code now. Stay tune.

Reply:
DONE PTAL

Reply:
Does not seem like this code got updated to use the m49 array?

Reply:
opps. you are right. need to change it.